### ISSUE TITLE
what's the difference between this project and CTranslate2 when using align

### ISSUE THREAD

**Issue by zyh3826** [guillaumekln/faster-whisper#192]
Hi, thanks for your work, it's really helpful!!!
I have a question about align.
when using faster-whisper, specific `word_timestamps=True` in `transcribe` method can obtain word timestamps, seems like `CTranslate2.whisper.align` can do the same thing, I wonder what's the difference between them.
By the way, are there any examples about CTranslate2.whisper.align, since I'm a new guy in ASR, the parameters really confuse me.
thanks a lot.

**Comment by guillaumekln** [comment]
Hi,

The method `Whisper.align` does not return the word timestamps. It returns some alignment information that is then used by faster-whisper to compute the word timestamps.

**Comment by zyh3826** [comment]
> Hi,
> 
> The method `Whisper.align` does not return the word timestamps. It returns some alignment information that is then used by faster-whisper to compute the word timestamps.

Hi guillaumekln, thanks for your reply.
I have another question, do you think finetuning whisper on my dataset can improve computing word timestamps？ I'm not sure the DTW algorithm has some relationships with the ASR results.
thanks a lot

**Comment by guillaumekln** [comment]
> do you think finetuning whisper on my dataset can improve computing word timestamps？

No.

**Comment by zyh3826** [comment]
> > do you think finetuning whisper on my dataset can improve computing word timestamps？
> 
> No.

Seems like need another algorithm like HMM

**Comment by anderleich** [comment]
Hi @guillaumekln ,

I'm using a custom Whisper model finetuned with the transformers library and properly converted to Ctranslate2. The model does not predict segment-level timestamps anymore as it is expected. 

So, my question is: are these segment-level timestamps as predicted by OpenAI's original models used in your implementation? If the finetuned models does not return segment level timestamps does it have a negative impact on the predicted word level timestamps?

**Comment by guillaumekln** [comment]
Yes, the predicted segment-level timestamps are used to determine how to advance the 30-second window over the input audio. faster-whisper implements the same logic as the original Whisper implementation.

Custom Whisper models are often fine-tuned without timestamps, so it can negatively impact the segment-level timestamps and by extension the word-level timestamps. However, I don't have much experience with custom models so the impact may vary from one model to another.

Note that the word-level timestamps accuracy also depends on the selected alignment heads. See also https://github.com/guillaumekln/faster-whisper/issues/338#issuecomment-1638145784.
